cmd/server/app/root.go

```diff
@@ -75,7 +96,7 @@ func initConfig() {
 	viper.SetConfigType("yaml")
 	viper.AutomaticEnv()
 
-	if err := viper.ReadInConfig(); err != nil {
+	if err = viper.ReadInConfig(); err != nil {
```

Comment:
This is not foolproof checking. Ideally, every time, we update the viper store to get values from the config file, using `ReadInConfig()` or any other method, we should first check the config file contents to prevent polluting the viper store.

Viper keeps track of the config file, a rare case, but the user can:

- Pass a correct config file
- Pollute it will null keys after the sanity check has been performed

If we do sort of a "hot-reload" of config file anywhere in the code by invoking appropriate viper methods, the store could end up having incorrect data. To fix it from our side, we should always perform a sanity check before reading the config file, directly or indirectly.

Reply:
We unmarshal the values from the viper store. Having a strongly typed struct won't prevent viper from auto updating or choosing the wrong values.

Like I mentioned in https://github.com/stacklok/minder/issues/2067#issuecomment-1877102081, if the key `http_server` has `null` value in yaml, it would be unmarshalled properly, but the viper store still can be corrupted. 

Having both keys `http_server` and `http_server.host` in viper config will give unpredictable results, we could resort to default even if we specified `http_server.host` by a flag. User passes the initial sanity checks and later modifies the yaml can cause problems.

> I think the race condition you're mentioning here is short

If by short you mean only exists in a short time window, then this isn't true for server. Functions invoked on server could read the config file again internally which can corrupt the viper store.